scripts/cluster-scanner/internal/search_old_clusters.go

```diff
@@ -0,0 +1,32 @@
+package internal
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/spectrocloud/palette-sdk-go/api/models"
+)
+
+// SearchOldClusters checks for clusters that have been running for more than 24 hours.
+// It takes a list of cluster summaries and returns a slice of messages with the
+// information of the clusters that were found.
+func SearchOldClusters(clusters []*models.V1SpectroClusterSummary) ([]string, error) {
+	// Slice of strings to keep track of any found clusters older than 24 hours
+	var messageArray []string
+
+	// Iterate through the clusters to find those running for more than 24 hours
+	for _, cluster := range clusters {
+		timeValue := time.Time(cluster.Metadata.CreationTimestamp)
+		clusterAge := time.Now().Sub(timeValue)
+
+		if clusterAge.Hours() >= 24 {
+			age, err := GetFormattedAge(clusterAge)
+			if err != nil {
+				return nil, err
+			}
+			message := fmt.Sprintf("The %s cluster named %s has been running for %s. Are you sure you need this cluster?", cluster.SpecSummary.CloudConfig.CloudType, cluster.Metadata.Name, *age)
```

Comment:
Food for thought. Say you have lots of clusters that are past 24 hrs in age. Do you want the output to include the "Are you sure you need this cluster?" in every line?

Reply:
Abs not :sweat_smile: Great point. Fixed it!